generate-api/src/main.rs

```diff
@@ -66,3 +68,32 @@ fn main() -> Result<()> {
 
     Ok(())
 }
+
+fn validate(objects: &mut Vec<Object>) {
+    validate_t_fmt_ampm(objects);
+}
+
+fn validate_t_fmt_ampm(objects: &mut Vec<Object>) {
+    for object in objects.iter_mut() {
+        if object.name == "LC_TIME" {
+            let mut found_t_fmt_ampm = false;
+            let mut am_pm_empty = false;
+            for (key, value) in object.values.iter() {
+                match (key.as_str(), value.as_slice()) {
+                    ("t_fmt_ampm" | "copy" | "insert", _) => found_t_fmt_ampm = true,
+                    ("am_pm", &[Value::String(ref am), Value::String(ref pm)]) => {
+                        am_pm_empty = am.is_empty() && pm.is_empty()
+                    }
+                    _ => {}
+                }
+            }
+            if !found_t_fmt_ampm {
+                let value = match am_pm_empty {
+                    true => vec![Value::String(String::new())],
```

Comment:
Should we still be pushing a value if `!am_pm_empty`? The logic here doesn't really make sense to me.